### DISCUSSION TITLE
begin as an afterword

### DISCUSSION THREAD

**Discussion by TheDoubleH** [microsoft/alguidelines#123] [BC Best Practices]
Discuss the `begin-as-an-afterword' best practice posted [here](https://alguidelines.dev/docs/bestpractices/begin-as-an-afterword/)

**Comment by rsaritzky** [comment]
I'm a bit older than most here, but when I first learned structured programming (learning ALGOL and COBOL), the BEGIN was always indented and on a separate line:

```
IF <Condition> THEN
  BEGIN
     <code>
  END

```
I was surprised to see the opposite in C/AL coding guidelines when I first started with NAV

And, in fact, I don't think the C/AL / AL guidelines cascade this into the CASE statement.  I don't recall anyone doing this:

```
CASE <VALUE> OF
  Value1: BEGIN
     <code>
    END;
  Value2: BEGIN
     <code>
  END;
END;
```

Isn't it always:

```
CASE <VALUE> OF
  Value1: 
    BEGIN
       <code>
    END;
  Value2: 
    BEGIN
     <code>
    END;
END;

```
?

**Comment by git-no** [comment]
In my point of view, one of the main priorities beside valid code should be the readability, because a user has to maintain the code.

## Case Statement

- It improves the readability of a `case` statement if `begin` and `end` are on separate lines.
- Beside this, if the command of the case is only online line it does not need a `begin` and `end` and is less code to read.

```al
case true of
  var1:
    <command>;
  var2:
    <command>;
end;
```

## If Statement

- One line ìf` statement should be without begin/end
 
```al
if <Condition> then
  <Command>;
```

- Regular if statement

```al
if <Condition> then begin
  <Command>;
  <Command>;
end;
```

- Multi condition if statement, especially if the condition is long

```al
if 
  <Condition> or 
  <Condition> or
  <Condition>
then begin
  <Command>;
  <Command>;
end;
```
